python/instrumentation/openinference-instrumentation-boto/src/openinference/instrumentation/boto/__init__.py

```diff
@@ -0,0 +1,169 @@
+            bound_arguments = call_signature.bind(*args, **kwargs)
+            bound_arguments.apply_defaults()
+
+            if bound_arguments.arguments.get("service_name") == "bedrock-runtime":
+                client = cast(InstrumentedClient, client)
+                client._unwrapped_invoke_model = client.invoke_model
+                client.invoke_model = _model_invocation_wrapper(tracer)(client)
+                return client
+            return client
+
+        return create_instrumented_client(*args, **kwargs)
+
+    return _client_wrapper  # type: ignore
+
+
+def _model_invocation_wrapper(tracer: Tracer) -> Callable[[InstrumentedClient], Callable[..., Any]]:
+    def _invocation_wrapper(wrapped_client: InstrumentedClient) -> Callable[..., Any]:
+        """Instruments a bedrock client's `invoke_model` method."""
+
+        @wraps(wrapped_client.invoke_model)
+        def instrumented_response(*args: Any, **kwargs: Any) -> Dict[str, Any]:
+            with tracer.start_as_current_span("bedrock.invoke_model") as span:
+                response = wrapped_client._unwrapped_invoke_model(*args, **kwargs)
+                response["body"] = BufferedStreamingBody(
+                    response["body"]._raw_stream, response["body"]._content_length
+                )
+                if raw_request_body := kwargs.get("body"):
+                    request_body = json.loads(raw_request_body)
+                response_body = json.loads(response.get("body").read())
+                response["body"].reset()
+
+                prompt = request_body.pop("prompt")
+                invocation_parameters = request_body
+
+                _set_span_attribute(span, SpanAttributes.LLM_PROMPTS, prompt)
+                _set_span_attribute(
+                    span, SpanAttributes.LLM_INVOCATION_PARAMETERS, invocation_parameters
+                )
+
+                if model_id := kwargs.get("modelId"):
+                    _set_span_attribute(span, SpanAttributes.LLM_MODEL_NAME, model_id)
+
+                    if isinstance(model_id, str):
+                        (vendor, _) = model_id.split(".")
+
+                    if vendor == "ai21":
+                        content = str(response_body.get("completions"))
+                        _set_span_attribute(
+                            span,
+                            MessageAttributes.MESSAGE_CONTENT,
+                            content,
+                        )
+                    elif vendor == "anthropic":
+                        content = str(response_body.get("completion"))
+                        _set_span_attribute(span, MessageAttributes.MESSAGE_CONTENT, content)
+                    elif vendor == "cohere":
+                        content = str(response_body.get("generations"))
+                        _set_span_attribute(
+                            span,
+                            MessageAttributes.MESSAGE_CONTENT,
+                            content,
+                        )
```

Comment:
nit. can be more succinct
```suggestion
                    if vendor == "ai21":
                        content = str(response_body.get("completions"))
                    elif vendor == "anthropic":
                        content = str(response_body.get("completion"))
                    elif vendor == "cohere":
                        content = str(response_body.get("generations"))
                    else:
                        content = "
                    if content:
                        _set_span_attribute(
                            span,
                            MessageAttributes.MESSAGE_CONTENT,
                            content,
                        )
```